### ISSUE TITLE
How to get list of JSON Pointers that have the format keyword of date-time value from a JSON Schema

### ISSUE THREAD

**Issue by puffyqi** [gregsdennis/json-everything#666]
### Documentation

- [X] I have consulted the [documentation](https://docs.json-everything.net), and my question isn't answered there.

### Nuget Package

JsonSchema.Net

### Package Version

6.0.3

### How can I help?  Please provide as much context as possible.

Hi,

I have some properties that have the format: date-time in the JSON Schema but I want to get those properties JSON pointer so that I can look for the value in JSON data.

Is there a way to do it?

Thank you.

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by gregsdennis** [comment]
I would use JSON Path.

You'll need to have the schema in `JsonNode` form.  Then use _JsonPath.Net_ with the path `$..[?@.format=='date-time'].format`.

Each result contains the node and the JSON Path to it.  Select the paths and you can convert them to pointer strings with the [`.AsJsonPointer()` extension](https://docs.json-everything.net/api/JsonPath.Net/JsonPathExtensions/#asjsonpointerthis-jsonpath-path).

**Comment by gregsdennis** [comment]
Oh! Also note that this won't catch `format` keywords in the schema root.  The `..` operation in JSON Path can't see the local node, only descendants.

**Comment by gregsdennis** [comment]
@puffyqi has this answered your question?

**Comment by puffyqi** [comment]
> @puffyqi has this answered your question?

Thanks for your help. I have tried and is able to work.

How I do:

- Get all the pointers from a JSON Schema file:

JsonNode schemanodes = JsonNode.Parse(schemadata);
JsonPath jpath = JsonPath.Parse("$..[?@.format=='date-time']");
PathResult pathresults = jpath.Evaluate(schemanodes);
NodeList nodeslist = pathresults.Matches;
 for (int i = 0; i < nodeslist.Count; i++)
       {
                string pointer = nodeslist[i].Location.AsJsonPointer();
                pointer_list.Add(pointer, new List<string>());
      }

- Get the pointers (instanceLocation) from Evaluation Results with the help of above pointers:

string schemalocationnode = node["schemaLocation"].ToString();

string instancelocationnode = node["instanceLocation"].ToString();

foreach (KeyValuePair<string, List<string>> entry in pointer_list)
                {
                    string pointername = entry.Key;

                    if (schemalocationnode.EndsWith(pointername))
                    {
                        List<string> datapointerslist = pointer_list[pointername];
                        datapointerslist.Add(instancelocationnode);
                    }
                }

- Get the data values from datapointerslist (list of instancelocation) Using JPointer

foreach (KeyValuePair<string, List<string>> entry in pointer_list)
            {
                List<string> pointers = entry.Value;
                
                if (pointers.Count > 0)
                {
                    foreach (string pointerlocation in pointers)
                    {
                        JsonPointer pointer = JsonPointer.Parse(pointerlocation);
                        JsonElement? element = pointer.Evaluate(jsondatadoc.RootElement);
                        if (element == null)
                        {
                            Console.WriteLine("cannot find pointer location: " + pointerlocation);
                        }
                        else
                        {
                            Console.WriteLine("data value: " + element.ToString());
                            Console.WriteLine("pointer: " + pointerlocation);
                        }
                    }
                }
                

            }

In Summary,

1. Get all the pointers in a JSON schema file where it have the format "date-time"
2. When verify schema file with data file, use the EvaluationResults and loop the details array. In details array, look for those "schemaLocation" that contains the pointers and save its "instanceLocation" in a list
3. Convert JSON data file to JSONDocument and loop the "instanceLocation" list to get the data values from each pointer in the list
